v0/destinations/ga/deleteUsers.js

```diff
@@ -0,0 +1,80 @@
+const { isEmpty } = require("lodash");
+const { httpPOST } = require("../../../adapters/network");
+const ErrorBuilder = require("../../util/error");
+const { executeCommonValidations } = require("../../util/regulation-api");
+const { GA_USER_DELETION_ENDPOINT } = require("./config");
+const { gaResponseHandler } = require("./utils");
+
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ *
+ * @param {*} userAttributes Array of objects with userId, emaail and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @param {Record<string, any> | undefined} rudderDestInfo contains information about the authorisation details to successfully send deletion request
+ * @returns
+ */
+const userDeletionHandler = async (userAttributes, config, rudderDestInfo) => {
+  const { secret } = rudderDestInfo;
+  // TODO: Should we do more validations ?
+  if (secret && isEmpty(secret)) {
+    throw new ErrorBuilder()
+      .setMessage(
+        // This would happen when server doesn't send "x-rudder-dest-info" header
+        // Todo's in-case this exception happen:
+        // 1. The server version might be an older one
+        // 2. There would have been some problem with how we are sending this header
+        `The "secret" field is not sent in "x-rudder-dest-info" header`
+      )
+      .setStatus(500)
+      .build();
+  }
+  await Promise.all(
+    userAttributes.map(async userAttribute => {
+      if (!userAttribute.userId) {
+        throw new ErrorBuilder()
+          .setMessage("User id for deletion not present")
+          .setStatus(400)
+          .build();
+      }
```

Comment:
Throwing an error inside the async function of `Promise.all` doesn't go well. As a result, we'd might declare the whole batch as an error where it could have been partially successful or wait forever for the promise to resolve.

Reply:
Can you double-check that?